File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.java
class Solution {
    public String reverseWords(String s) 
    {
        s = s.strip();
        String res[] = s.split(" ");
        String r = "";
        for(int i = 0 ; i < res.length ; i++)
            System.out.println(res[i].length());
        for(int i = res.length - 1; i > 0 ; i--)
        {
            // int check = 0;
            // for(int j = 0 ; j < res[i].length() ; j++)
            // {
            //     if(res[i].charAt(j) == ' ')
            //     {
            //         System.out.println("ji");
            //         check = 1;
            //     }
            //     else
            //     {
            //         check = 0;
            //         break;
            //     }
            // }
            // if(check == 0)
            // {
            //     r += res[i];
            //     r += ' ';
            // }
            if(res[i].length() == 0)
                continue;
            else
                r += res[i];
                r += ' ';
        }
        r += res[0];
        return r;
        
    }
}

File: 0151-reverse-words-in-a-string/0151-reverse-words-in-a-string.cpp
class Solution {
public:
    string reverseWords(string s) 
    {
        if(s.size() == 0)
            return "";
        int i = 0;
        while(i < s.size() && s[i] == ' ')
            i++;
        if(i == s.size())
            return "";
        int j = i;
        while(j < s.size() && s[j] != ' ')
            j++;
        
        string word = s.substr(i , j - i);
        
        string res = reverseWords(s.substr(j));
        
        if(res == "")
            return word;
        return res + " " + word;
        
    }
};